testing/cypress/integration/category/category.spec.js

```diff
@@ -0,0 +1,128 @@
+        cy.get(`[data-background-image='/documents/category/16/living-room.jpg']`)
+            .invoke("attr", "style", "background-image: url('/documents/category/16/living-room.jpg')")
+            .should("have.attr", "style", "background-image: url('/documents/category/16/living-room.jpg')");
+    });
+
+    it("should be able to navigate by sidebar", () =>
+    {
+        cy.get(".widget-inner > :nth-child(3) > .nav-item > .nav-link > .d-flex > .flex-grow-1").click();
+        cy.location("pathname").should("eq", "/arbeitszimmer-buero/");
+    });
+
+    it("should be able to apply min price filter at category toolbar", () =>
+    {
+        cy.getByTestingAttr("category-toolbar-filter").click();
+        cy.get(".widget-filter-price").find("input").first().type("400", { delay: 30 });
+        cy.get(".widget-filter-price").find("button").click();
+        cy.get(".product-list").children().first()
+            .find(".thumb-content").find(".price").should("contain", "400,00");
+    });
+
+    it("should be able to apply max price filter at category toolbar", () =>
+    {
+        cy.getByTestingAttr("category-toolbar-filter").click();
+        cy.get(".widget-filter-price").find("input").last().type("0.01", { delay: 30 });
+        cy.get(".widget-filter-price").find("button").click();
+        cy.get(".product-list").children().first()
+            .find(".thumb-content").find(".price").should("contain", "0,01");
+    });
+
+    it("should be able to apply attribute filter at category toolbar", () =>
+    {
+        cy.getByTestingAttr("category-toolbar-filter").click();
+        cy.get(".widget-filter-attributes-properties-characteristics").find(".option-1").click();
+        cy.get(".widget-selected-filter").find(".selected-filters").children().first().should("exist");
+        cy.get(".product-list").find("li").should("have.length.gt", 0);
+    });
+
+    it("should remove applied attribute filter at category toolbar", () =>
+    {
+        cy.getByTestingAttr("category-toolbar-filter").click();
+        cy.get(".widget-filter-attributes-properties-characteristics").find(".option-1").click();
+        cy.getByTestingAttr("category-toolbar-close").click();
+        cy.get(".widget-selected-filter").find(".selected-filters").children().first().click();
+        cy.get(".product-list").find("li").should("have.length", 20);
+    });
+
+    it("should list items for category", () =>
+    {
+        cy.get(".product-list").find("li").should("have.length", 20);
+    });
+
+    it("should click card and open item on single item", () =>
+    {
+        cy.get(".product-list").find("li").first().scrollIntoView().find(".thumb-title").click();
+        cy.location("pathname").should("eq", "/wohnzimmer/sessel-hocker/barhocker-white-sanfrancisco_109_1007/");
+    });
+
+    it("should add item to basket when button is clicked on page item", () =>
+    {
+        cy.get(".product-list").find("li").first().find(".add-to-basket-lg-container").click();
```

Comment:
Würde bedeuten wir gehen immer davon aus das ein Artikel in den Warenkorb legbar ist ?

Bei Variantenartikeln ist das nicht der Fall. Können wir sicherstellen das die Reihenfolge sich nicht ändert ?

Reply:
done hab die id genommen